### PR TITLE
AMPI: Use ROMIO's MPI_File_{get,set}_errhandler instead of stubs

### PR DESCRIPTION
**Pull request by evan-charmworks**
Please **rebase** as a merge strategy.

### CODE REVIEW

**Comment by evan-charmworks** [comment]
Ready for review

**Review by stwhite91 — APPROVED** [review]

File: src/libs/ck-libs/ampi/ampi.C [hunk 1]
```diff
@@ -10002,6 +10002,8 @@ AMPI_API_IMPL(int, MPI_File_create_errhandler, MPI_File_errhandler_function *fun
   return MPI_SUCCESS;
 }
 
+#if !CMK_AMPI_WITH_ROMIO
+// Disable ROMIO's get_errh.c and set_errh.c when implementing these.
 AMPI_API_IMPL(int, MPI_File_set_errhandler, MPI_File file, MPI_Errhandler errhandler)
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/ampi.C:10007]: Why are we implementing these functions at all?

Comment (evan-charmworks) [reply src/libs/ck-libs/ampi/ampi.C:10007]: It seems like `errhandler` code is intended to be shared between MPI_File, MPI_Comm, and MPI_Win. ROMIO has this:

```cpp
/* Section 4.13.3 */
#ifndef MPICH
/* MPICH provides these definitions */
AMPI_CUSTOM_FUNC(int, MPI_File_set_errhandler, MPI_File file, MPI_Errhandler errhandler)
AMPI_CUSTOM_FUNC(int, MPI_File_get_errhandler, MPI_File file, MPI_Errhandler *errhandler)
#endif
```

No other `*_errhandler` functions are present in ROMIO.

Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampi.C:10007]: Right, but is having these no-op functions in the AMPI/no ROMIO case useful at all?

Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampi.C:10007]: Ok, let's leave them there for now.

**Review by matthiasdiener — APPROVED** [review]

**Pull request merged**